src/features/Posts/postsSlice.js

```diff
@@ -34,13 +44,31 @@ const postsSlice = createSlice({
 				} else {
 					state.status = "succeeded";
 				}
+				if(state.searchTerm !== action.payload.searchTerm || state.subreddit !== action.payload.subreddit){
+					state.listing = []
```

Comment:
Can you explain your reasoning behind this? I included this so that in the unlikely event that we see all of the posts on the front page we get a "Nothing left to see here message"